packages/block/src/block.ts

```diff
@@ -63,8 +63,15 @@ export class Block {
     }
 
     // parse transactions
+    const common = Object.create(this._common)
+    const blockNumber = parseInt(this.header.number.toString('hex'), 16)
+    common.setHardforkByBlockNumber(blockNumber)
+
```

Comment:
This is now [done](https://github.com/ethereumjs/ethereumjs-vm/blob/master/packages/block/src/header.ts#L146) in the header constructor (I tried to deduplicate on functionality executed in the one or the other constructor where ever possible) and therefore redundant here.

This direct setting would also undermine the new `hardforkByBlockNumber` `BlockOption` introduced in #863 . Reasoning to introduce this option is that we want to provide both possibilities for a hardfork either be kept as set (e.g. when a block is created from within the VM running on specific HF settings regardless of the block number) to have a consistent execution along with the possibility to instantiate a block which should behave regarding to the block number set (e.g. in the client context, but likely for other use cases as well).

Reply:
resolved in https://github.com/ethereumjs/ethereumjs-vm/pull/812/commits/bc0e0a8be2c8adfb78e926cbc140dc8985491611